### DISCUSSION TITLE
Why the data shape are different through transform "spacingd.

### DISCUSSION THREAD

**Discussion by xxsxxsxxs666** [Project-MONAI/MONAI#5934] [General]
![1675428782920](https://user-images.githubusercontent.com/61532031/216608306-d48a11ff-d351-47a3-af81-ffd5f01b98c9.jpg)

    val_transforms = Compose(
        [
            LoadImaged(keys=["image", "label"]),
            EnsureChannelFirstd(keys=["image", "label"]),
            ScaleIntensityRanged(
                keys=["image"], a_min=-57, a_max=400,
                b_min=0.0, b_max=1.0, clip=True,
            ),
            CropForegroundd(keys=["image", "label"], source_key="image"),  # default: value>0
            Orientationd(keys=["image", "label"], axcodes="RAI"),
            Spacingd(keys=["image", "label"], pixdim=(
                0.35, 0.35, 0.5), mode=("bilinear", "nearest")),  # Now only Sequential_str, How to add spline
        ]
    )

The shape and metadata are the same between image and label
![image](https://user-images.githubusercontent.com/61532031/216610903-9b586e06-0dd5-4cb4-85d8-05904146b159.png)

this problem only happen when I use cachedataset. Nothing went wrong when I use normal dataset.

**Comment by wyli** [comment]
that looks like a bug and at the moment please consider using `EnsureSameShaped` as a workaround:

https://github.com/Project-MONAI/MONAI/blob/a77de4adff49037ffb24ea2cae217fac40302181/monai/apps/auto3dseg/transforms.py#L25-L28
